### PR TITLE
update!: bump libs to latest and deprecation of k8s metadata options and configs

### PR DESCRIPTION
**Pull request by jasondellaluce**
**What type of PR is this?**

/kind cleanup

**Any specific area of the project related to this PR?**

/area build

/area engine

**What this PR does / why we need it**:

This bumps the libs dependency (which we haven't done in a while), and inherently deprecates the `-k`, `-K`, `--k8s-node` options of Falco that are no more functional (alongside all config options related to k8s) as tracked in https://github.com/falcosecurity/falco/issues/2763.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
update!: bump libs to latest and deprecation of k8s metadata options and configs
```

### CODE REVIEW

File: falco.yaml [hunk 1]
```diff
@@ -170,11 +167,10 @@ rules_file:
 #
 # Please note that if your intention is to enrich Falco syscall logs with fields
 # such as `k8s.ns.name`, `k8s.pod.name`, and `k8s.pod.*`, you do not need to use
```
Comment (Andreagit97) [inline falco.yaml:169]: yes the container fields will have another name but I will update the doc in another PR

**Review by Andreagit97 — APPROVED** [review]
Thank you!
/approve

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**